drivers/input/touchscreen/himax.c

```diff
@@ -0,0 +1,462 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * Driver for Himax touchscreens
+ *
+ * Copyright (C) 2022 Job Noorman <job@noorman.info>
+ *
+ * This code is based on "Himax Android Driver Sample Code for QCT platform":
+ *
+ * Copyright (C) 2017 Himax Corporation.
+ */
+
+#include <linux/kernel.h>
+#include <linux/err.h>
+#include <linux/delay.h>
+#include <linux/interrupt.h>
+#include <linux/i2c.h>
+#include <linux/gpio/consumer.h>
+#include <linux/input.h>
+#include <linux/input/mt.h>
+#include <linux/input/touchscreen.h>
+#include <asm/byteorder.h>
+
+#define HIMAX_GPIO_RST_NAME "reset"
+
+#define HIMAX_ID_83112A 0x83112a
+#define HIMAX_ID_83112B 0x83112b
+
+#define HIMAX_MAX_POINTS 10
+
+#define HIMAX_CMD_REG_SET_ADDR 0x00
+#define HIMAX_CMD_REG_INIT_READ 0x0c
+#define HIMAX_CMD_REG_READ_VALUE 0x08
+#define HIMAX_CMD_READ_EVENT 0x30
+
+#define HIMAX_REG_PRODUCT_ID 0x900000d0
+
+struct himax_event_point {
+	__be16 x;
+	__be16 y;
+} __packed;
+
+struct himax_event {
+	struct himax_event_point points[HIMAX_MAX_POINTS];
+	u8 majors[HIMAX_MAX_POINTS];
+	u8 pad0[2];
+	u8 num_points;
+	u8 pad1[2];
+	u8 checksum_fix;
+} __packed;
+
+static_assert(sizeof(struct himax_event) == 56);
+
+struct himax_ts_data {
+	struct gpio_desc *gpiod_rst;
+	struct input_dev *input_dev;
+	struct i2c_client *client;
+	struct touchscreen_properties props;
+};
+
+static int himax_i2c_read(struct himax_ts_data *ts, u8 command, u8 *data,
+			  u8 length)
+{
+	int ret;
+	struct i2c_client *client = ts->client;
+
+	// clang-format off
```

Comment:
get rid of the clang-format comments throughout the file

Reply:
I'll do this as a last step. The problem is that the kernel's clang-format config completely messes up some struct initializers. I don't like adding these guards in the code either but I haven't found a better way to make it ignore them.

Reply:
I removed the comments. I did add some trailing commas after the sentinel values in the struct initializers as this also convinces `clang-format` to not mess up. I hope that's ok.